src/main/java/org/zephyrsoft/sdb2/service/IndexerService.java

```diff
@@ -69,6 +70,8 @@
 	
 	private final ExecutorService executor = Executors.newSingleThreadExecutor();
 	
+	private final CopyOnWriteArrayList<OnIndexChangeListener> onIndexChangeListeners = new CopyOnWriteArrayList<>();
```

Comment:
Please use only `List`as variable type.

BTW: Is there a reason why a normal `ArrayList` wouldn't suffice? Are listeners added/removed from multiple threads in parallel?

Reply:
No, you are right. There is no parallel access. I think I just copied it somwhere.. Changed it to ArrayList.